Sources/NIOHTTP1Server/main.swift

```diff
@@ -33,6 +33,28 @@ extension String {
     }
 }
 
+private func httpResponseHead(request: HTTPRequestHead, status: HTTPResponseStatus, headers: HTTPHeaders = HTTPHeaders()) -> HTTPResponseHead {
+    var head = HTTPResponseHead(version: request.version, status: status, headers: headers)
+    let connectionHeaders: [String] = head.headers[canonicalForm: "connection"].map { $0.lowercased() }
```

Comment:
Is this type annotation necessary?

Reply:
not necessary

Reply:
What else would it be, given that we compare its contents to strings everywhere else?

Reply:
a `String`. `someString.contains("keep-alive")` isn't that crazy. Sure, it'd be bad but I like the type annotation there. Can obvs remove if you think it makes it less readable